src/app/file-upload/file-upload.component.ts

```diff
@@ -22,39 +22,88 @@ export class FileUploadComponent {
 
   constructor() { }
 
-  @HostListener('dragover', ['$event']) onDragOver(event) {
+  /**
+   * Drag/drop event handler to apply styles when items are dragged/dropped. This gives
+   * the user feedback when using drag/drop in the file picker UI.
+   *
+   * @param {DragEvent} event A drag over event.
+   * @return {void}
+   */
+  @HostListener('dragover', ['$event']) onDragOver(event: DragEvent): void {
     this.dragDropClass = 'droparea';
     event.preventDefault();
   }
 
-  @HostListener('dragenter', ['$event']) onDragEnter(event) {
+  /**
+   * Drag/drop event handler to apply styles when items are dragged/dropped. This gives
+   * the user feedback when using drag/drop in the file picker UI.
+   *
+   * @param {DragEvent} event A drag over event.
+   * @return {void}
+   */
+  @HostListener('dragenter', ['$event']) onDragEnter(event: DragEvent): void {
     this.dragDropClass = 'droparea';
     event.preventDefault();
   }
 
-  @HostListener('dragend', ['$event']) onDragEnd(event) {
+  /**
```

Comment:
All of these identical docs - could they be slightly more specific  à la onDrop?

Reply:
The functionality within each of these functions is the same and only the particular event changes so I felt it was reasonable to have the same description. I'll update it to reference specific drag vs drop events in a future PR